src/compile_commands.cc

```diff
@@ -0,0 +1,79 @@
+#include "compile_commands.h"
+#include <boost/property_tree/json_parser.hpp>
+
+std::vector<std::string> CompileCommands::Command::paramter_values(const std::string &paramter_name) const {
+  std::vector<std::string> parameter_values;
+  
+  bool found_argument=false;
+  for(auto &parameter: parameters) {
+    if(found_argument) {
+      parameter_values.emplace_back(parameter);
+      found_argument=false;
+    }
+    else if(parameter==paramter_name)
+      found_argument=true;
+  }
+  
+  return parameter_values;
+}
+
+CompileCommands::CompileCommands(const boost::filesystem::path &build_path) {
+  try {
+    boost::property_tree::ptree root_pt;
+    boost::property_tree::json_parser::read_json((build_path/"compile_commands.json").string(), root_pt);
+    
```

Comment:
Correct me if I'm wrong, but I think read_json can cause problems. I think there is a error_code version.